src/provider.tsx

```diff
@@ -64,12 +64,29 @@ class LDProvider extends React.Component<ProviderConfig, HocState> implements En
     });
   };
 
-  async componentDidMount() {
-    const { clientSideID, user, flags, options } = this.props;
+  initLDClient = async () => {
+    const { clientSideID, flags, options, user } = this.props;
     const reactOptions = this.getReactOptions();
     const { flags: fetchedFlags, ldClient } = await initLDClient(clientSideID, user, reactOptions, options, flags);
     this.setState({ flags: fetchedFlags, ldClient });
     this.subscribeToChanges(ldClient);
+  };
+
+  async componentDidMount() {
+    const { user, deferInitialization } = this.props;
+    if (deferInitialization && !user) {
+      return;
+    }
+
+    return this.initLDClient();
```

Comment:
I think `initLDClient` needs to be awaited, since it's an async function.

Reply:
Hmm I think this should be fine since an async function always returns a promise. It will be up to the consumer to await it if it needs the return value.

Reply:
There is a lint rule that forces all functions that return a Promise to be `async` functions. That's why I've labeled them as `async`.

Reply:
@bezreyhan the previous behavior in `provider.tsx` is to await this initialization in componentDidMount. We would like to keep this behavior consistent please if we can. Furthermore, I don't think it is possible to await a promise from a react's lifecycle method like componentDidMount.

Reply:
@yusinto I just tried to `await` the return value but got the following lint error: `Unnecessary 'await'.tslint(no-return-await)`.

I believe since an async function always returns a Promise, the behavior should not change if you `await` the return value or not.  Also it might be preferable not to await it because you will then need to add error handling, but I'm not 100% sure about this.

Reply:
@yusinto I think I may have misunderstood you. I just updated my PR to `await` initialization and I am no longer returning anything. Please let me know if this is sufficient.